core/base/mergeTreePrincipalGeodesics/MergeTreePrincipalGeodesics.h

```diff
@@ -0,0 +1,1652 @@
+        getInterpolation<dataType>(
+          barycenter, vS_[i], v2s_[i], 0.0, extremityV1);
+        getInterpolation<dataType>(
+          barycenter, vS_[i], v2s_[i], 1.0, extremityV2);
+        // Get distance
+        dataType distance;
+        computeOneDistance(
+          extremityV1, extremityV2, distance, true, useDoubleInput_);
+        if(trees2NodeCorr_.size() != 0) {
+          ftm::MergeTree<dataType> extremity2V1, extremity2V2;
+          getInterpolation<dataType>(
+            barycenter2, trees2Vs_[i], trees2V2s_[i], 0.0, extremity2V1);
+          getInterpolation<dataType>(
+            barycenter2, trees2Vs_[i], trees2V2s_[i], 1.0, extremity2V2);
+          // Get distance
+          dataType distance2;
+          computeOneDistance(extremity2V1, extremity2V2, distance2, true,
+                             useDoubleInput_, false);
+          distance = mixDistances(distance, distance2);
+        }
+        // std::cout << distance << std::endl;
+        for(unsigned int j = 0; j < allTs_[i].size(); ++j)
+          allScaledTs_[i][j] = allTs_[i][j] * distance;
+      }
+    }
+
+    template <class dataType>
+    void computeBranchesCorrelationMatrix(
+      ftm::MergeTree<dataType> &barycenter,
+      std::vector<ftm::MergeTree<dataType>> &trees) {
+      branchesCorrelationMatrix_ = std::vector<std::vector<double>>(
+        barycenter.tree.getNumberOfNodes(),
+        std::vector<double>(numberOfGeodesics_, 0.0));
+      persCorrelationMatrix_ = branchesCorrelationMatrix_;
+
+      // m[i][j] contains the node in trees[j] matched to the node i in the
+      // barycenter
+      std::vector<std::vector<ftm::idNode>> matchingMatrix;
+      getMatchingMatrix(barycenter, trees, baryMatchings_, matchingMatrix);
+
+      std::queue<ftm::idNode> queue;
+      queue.emplace(barycenter.tree.getRoot());
+      while(!queue.empty()) {
+        ftm::idNode node = queue.front();
+        queue.pop();
+
+        // Get births and deaths array
+        std::vector<double> births(trees.size(), 0.0),
+          deaths(trees.size(), 0.0), pers(trees.size(), 0.0);
+        for(unsigned int i = 0; i < trees.size(); ++i) {
+          auto matched = matchingMatrix[node][i];
+          std::tuple<dataType, dataType> birthDeath;
+          if((int)matched == -1) {
+            birthDeath = barycenter.tree.template getBirthDeath<dataType>(node);
+            auto projec
+              = (std::get<0>(birthDeath) + std::get<1>(birthDeath)) / 2.0;
+            birthDeath = std::make_tuple(projec, projec);
+          } else
+            birthDeath
+              = trees[i].tree.template getBirthDeath<dataType>(matched);
+          births[i] = std::get<0>(birthDeath);
+          deaths[i] = std::get<1>(birthDeath);
+          pers[i] = deaths[i] - births[i];
+        }
+
+        // Compute correlations
+        for(unsigned int g = 0; g < numberOfGeodesics_; ++g) {
+          double birthCorr = Statistics::corr(births, allTs_[g]);
```

Comment:
could you please change to `ttk::Statistics::corr` to make it more explicit.
could you please double-check that in the rest of the code?

Reply:
Done in the next commit!